### PR TITLE
Optimize serializing JSON messages.

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt [hunk 1]
```diff
@@ -326,6 +359,17 @@ class ForwardedEndpointsMessage(
     val endpointsEnteringLastN: Collection<String>,
     val conferenceEndpoints: Collection<String>
 ) : BridgeChannelMessage(TYPE) {
+    /**
+     * Serialize using json-simple because it's faster.
+     */
+    override fun toJson(): String {
```
Comment (bbaldino) [inline jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt:365]: use the `= JsonObject().apply {` style like above?

**Comment by bgrozev** [comment]
With the code from the second commit on my machine I get:
```
Times=1000000
Jackson: 11584
Json-simple: 386
String concat: 41
String template: 43
Raw string template: 3322
```

File: jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt [hunk 2]
```diff
@@ -338,6 +379,12 @@ class ForwardedEndpointsMessage(
  * TODO: consider and adjust the format of videoConstraints. Do we need all of the VideoConstraints fields? Document.
  */
 class SenderVideoConstraintsMessage(val videoConstraints: VideoConstraints) : BridgeChannelMessage(TYPE) {
+    /**
+     * Serialize manually because it's faster than either Jackson or json-simple. Note that kotlin raw strings (triple
+     * quotes) are also significantly slower.
+     */
+    override fun toJson(): String = "{\"colibriClass\":\"$TYPE\", \"videoConstraints\":$videoConstraints}"
```
Comment (JonathanLennox) [inline jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt:386]: Add a comment to `VideoConstraints.toString()` that we depend on the fact that it returns valid JSON?

**Review by bbaldino — APPROVED** [review]

File: jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt [hunk 3]
```diff
@@ -338,6 +377,12 @@ class ForwardedEndpointsMessage(
  * TODO: consider and adjust the format of videoConstraints. Do we need all of the VideoConstraints fields? Document.
  */
 class SenderVideoConstraintsMessage(val videoConstraints: VideoConstraints) : BridgeChannelMessage(TYPE) {
+    /**
+     * Serialize manually because it's faster than either Jackson or json-simple.
+     * Note that we depend on `VideoConstraints.toString` producing JSON.
```
Comment (JonathanLennox) [inline jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt:382]: I actually meant to put the comment in `VideoConstraints`, not here -- but if you say breaking that will cause the tests to fail, that's fine.

**Comment by bbaldino** [comment]
Out of curiosity, I also tried `StringBuilder`:
```
fun toJsonStringBuilder(m: DominantSpeakerMessage) = buildString {
    append("{")
    append("\"colibriClass\": \"${DominantSpeakerMessage.TYPE}\",")
    append("\"dominantSpeakerEndpoint\": \"${m.dominantSpeakerEndpoint}\"")
    append("}")
}
```
It falls in between Json-simple and Raw string template:
```
Times=1000000
Jackson: 16090
Json-simple: 490
String concat: 54
String template: 52
Raw string template: 49
Raw string template (trim): 5245
String builder: 2902
```

**Pull request merged**